src/ssl.c

```diff
@@ -275,6 +277,482 @@ const WOLF_EC_NIST_NAME kNistCurves[] = {
 };
 #endif
 
+#if defined(HAVE_HPKE) && defined(HAVE_ECC)
+int wolfSSLGenerateEchConfig(WOLFSSL_CTX* ctx, const char* publicName)
+{
+    int ret = 0;
+    int rng_ret = 0;
+    word16 encLen = 32;
+#ifdef WOLFSSL_SMALL_STACK
+    Hpke* hpke;
+    WC_RNG* rng;
+#else
+    Hpke hpke[1];
+    WC_RNG rng[1];
+#endif
+
+    if (ctx == NULL || publicName == NULL)
+      return BAD_FUNC_ARG;
+
+    ctx->echConfigs = (EchConfig*)XMALLOC(sizeof(EchConfig), ctx->heap,
+        DYNAMIC_TYPE_NONE);
+    ForceZero(ctx->echConfigs, sizeof(EchConfig));
+
+    if (ctx->echConfigs == NULL)
+        return MEMORY_E;
+
+#ifdef WOLFSSL_SMALL_STACK
+    hpke = MALLOC(sizeof(Hpke), ctx->heap, DYNAMIC_TYPE_NONE);
```

Comment:
Should be `XMALLOC`. Please test your build with `./configure --enable-hpke --enable-small-stack`.